### PR TITLE
Make the testopts accessible in your tests.

### PR DESCRIPTION
**Pull request by aiomaster**
Hello,

I would like to have the options available to use in my tests.
Especially the seed value is very interesting for me.
I would like to make some tests use different, random but reproducable input everytime the tests run to cover more test cases.
The seed value is good to support that kind of stuff, but I have to access it somehow in my test cases.
By exposing the options as attr_accessor I can use it.

Thanks!

### CODE REVIEW

**Comment by zenspider** [comment]
For my own understanding... You want to use `srand` or something with a random but reproducible value?

File: lib/minitest.rb [hunk 1]
```diff
@@ -46,6 +46,10 @@ module Minitest
   self.info_signal = "INFO"
 
   ##
+  # Access the testopts for example to use the seed value in your tests.
+  mc.send :attr_accessor, :testopts
```
Comment (zenspider) [inline lib/minitest.rb]: "testopts" isn't a word, nor are they specific to testing. Please switch this to "option" (EDIT: singular).

Comment (aiomaster) [inline lib/minitest.rb]: Why singular and not "options"? There is more than one option in that hash.

**Comment by aiomaster** [comment]
You are right, I basically want to use srand with this seed value to set a reproducable but random state for my tests.
In combination for example with the Faker gem (https://github.com/stympy/faker) which uses internally the random class of ruby to generate different types of data you can have more powerfull tests that don't test exactly the same input again and again, but more cases where you couldn't think of when creating your tests. But it is very important, that you can reproduce the values in case of a failure.

**Comment by sandstrom** [comment]
friendly ping @aiomaster @zenspider  😄 

This would be very useful, I'd also like to access the seed value.

**Comment by aiomaster** [comment]
I rebased from current master.

File: lib/minitest.rb (hunk 1 above)
Comment (zenspider) [reply lib/minitest.rb]: You only access them one at a time.

**Comment by andrelaszlo** [comment]
We're also using the Faker gem, and would like to seed it with the same seed as Minitest - there are some intermittent failures that are really annoying to track down otherwise.

This feature would save us a lot of time, @zenspider! Is there something you disagree with in this implementation? I'd be happy to fix it if @aiomaster isn't available.

**Comment by zenspider** [comment]
I absolutely do NOT want options available to tests. Instead, I've exposed the seed being used via `Minitest.seed`.

**Pull request closed without merging**